test/unit/puppetlabs/services/ca/certificate_authority_core_test.clj

```diff
@@ -119,6 +119,44 @@
           response (ring-app request)]
       (is (= version-number (get-in response [:headers "X-Puppet-Version"]))))))
 
+(deftest handle-delete-certificate-request!-test
+  (let [settings      (assoc (testutils/ca-sandbox! cadir)
+                             :allow-duplicate-certs true
+                             :autosign false)
+        subject       "foo-agent"
+        csr-stream    (io/input-stream (test-pem-file "foo-agent-csr.pem"))
+        expected-path (ca/path-to-cert-request (:csrdir settings) subject)
+        assert-proper
+          (fn
+            [response code matcher log-level]
+            (do
+             (is (= code (:status response)))
+             (is (re-matches matcher (:body response)))
+             (is (= "text/plain" (get-in response [:headers "Content-Type"])))
+             (is (logged? matcher log-level))))]
+    (logutils/with-test-logging
+      (testing "Successful CSR deletion"
+        (try
+          (handle-put-certificate-request! subject csr-stream settings)
+          (is (true? (fs/exists? expected-path)))
+          (let [response (handle-delete-certificate-request! subject settings)
+                msg-matcher (re-pattern (str "Deleted .* for " subject ".*"))]
+            (is (false? (fs/exists? expected-path)))
+            (assert-proper response 204 msg-matcher :debug))
+          (finally
+            (fs/delete expected-path))))
+      (testing "Attempted deletion of a non-existant CSR"
+        (let [response (handle-delete-certificate-request! subject settings)
+              msg-matcher (re-pattern (str "No cert.* " subject " at .*"))]
+          (is (false? (fs/exists? expected-path)))
+          (assert-proper response 404 msg-matcher :info)))
+      (testing "Error during deletion of a CSR"
+        (let [response
+              (with-redefs [me.raynes.fs/exists? (constantly true)]
```

Comment:
I'm not sure why clj is still able to delete this file but I can't get changing perms on the file from clj to work:
```clj
user=> (fs/writeable? "dev-resources/puppetlabs/services/ca/certificate_authority_core_test/pems/foo-agent-csr.pem")
true
user=> (fs/chmod "-w" "dev-resources/puppetlabs/services/ca/certificate_authority_core_test/pems/foo-agent-csr.pem")
"dev-resources/puppetlabs/services/ca/certificate_authority_core_test/pems/foo-agent-csr.pem"
user=> (fs/writeable? "dev-resources/puppetlabs/services/ca/certificate_authority_core_test/pems/foo-agent-csr.pem")
false
user=> (fs/delete "dev-resources/puppetlabs/services/ca/certificate_authority_core_test/pems/foo-agent-csr.pem")
true
user=> (fs/exists? "dev-resources/puppetlabs/services/ca/certificate_authority_core_test/pems/foo-agent-csr.pem")
false
```
While in the other window:
```
15:11:25 hobo:pems justin (bug/master/SERVER-977_add_cert_request_delete *%) ‖ ls -l
total 104
...snip...
-rw-r--r--  1 justin  staff   985 Apr  5 15:11 foo-agent-csr.pem

15:11:28 hobo:pems justin (bug/master/SERVER-977_add_cert_request_delete *%) ‖ ls -l
total 104
...snip...
-r--r--r--  1 justin  staff   985 Apr  5 15:11 foo-agent-csr.pem


15:19:40 hobo:pems justin (bug/master/SERVER-977_add_cert_request_delete *%) ‖ ls | grep foo-agent
15:20:06 hobo:pems justin (bug/master/SERVER-977_add_cert_request_delete *%) ‖

```